sklearn/decomposition/tests/test_fastica.py

```diff
@@ -265,21 +319,34 @@ def test_fit_transform():
         (False, 10, (10, 10)),
     ],
 )
-def test_inverse_transform(whiten, n_components, expected_mixing_shape):
+def test_inverse_transform(
+    whiten, n_components, expected_mixing_shape, global_random_seed, global_dtype
+):
     # Test FastICA.inverse_transform
     n_samples = 100
-    rng = np.random.RandomState(0)
-    X = rng.random_sample((n_samples, 10))
+    rng = np.random.RandomState(global_random_seed)
+    X = rng.random_sample((n_samples, 10)).astype(global_dtype)
 
     ica = FastICA(n_components=n_components, random_state=rng, whiten=whiten)
-    Xt = ica.fit_transform(X)
+    with warnings.catch_warnings():
+        # For some dataset (depending on the value of global_dtype) the model
+        # can fail to converge but this should not impact the definition of
+        # a valid inverse transform.
+        warnings.simplefilter("ignore", ConvergenceWarning)
+        Xt = ica.fit_transform(X)
     assert ica.mixing_.shape == expected_mixing_shape
     X2 = ica.inverse_transform(Xt)
     assert X.shape == X2.shape
 
     # reversibility test in non-reduction case
     if n_components == X.shape[1]:
-        assert_array_almost_equal(X, X2)
+        # XXX: we have to set atol for this test to pass for all seeds when
+        # fitting with float32 data. Is this is a revealing a bug?
+        if global_dtype:
+            atol = np.abs(X2).mean() / 1e5
```

Comment:
Note that dividing by 1e6 would fail for some seeds here...

Reply:
```suggestion
            # XXX: dividing by a smaller number makes
            # tests fail for some seeds.
            atol = np.abs(X2).mean() / 1e5
```